### ISSUE TITLE
IFlowCaptureReferenceOperation.IsInitialization

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#57484]
## Background and Motivation

In the process of adding the API for IInterpolatedStringHandlerCreationOperation, we have a case in the lowering where a compiler temp (represented in the CFG by an `IFlowCaptureReferenceOperation`) is initialized in an `out` parameter position by a constructor call. We don't have a way to represent this information in the graph today, so we need a flag to capture that info. This API is similar to [`ILocalReferenceOperation.IsDeclaration`](https://sourceroslyn.io/#Microsoft.CodeAnalysis/Generated/Operations.Generated.cs,786).

## Proposed API

```diff
namespace Microsoft.CodeAnalysis.FlowAnalysis
{
     public class IFlowCaptureReferenceOperation
     {
+        public bool IsInitialization { get; }
     }
```

## Usage Examples

An object creation with the parameter looks like this:

``` 
                    IObjectCreationOperation (Constructor: CustomHandler..ctor(System.Int32 literalLength, System.Int32 formattedCount, out System.Boolean success)) (OperationKind.ObjectCreation, Type: CustomHandler, IsImplicit) (Syntax: '$""literal{c,1:format}""')
                      Arguments(3):
                          IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: literalLength) (OperationKind.Argument, Type: null, IsImplicit) (Syntax: '$""literal{c,1:format}""')
                            ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 7, IsImplicit) (Syntax: '$""literal{c,1:format}""')
                            InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                            OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                          IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: formattedCount) (OperationKind.Argument, Type: null, IsImplicit) (Syntax: '$""literal{c,1:format}""')
                            ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1, IsImplicit) (Syntax: '$""literal{c,1:format}""')
                            InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                            OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                          IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: success) (OperationKind.Argument, Type: null, IsImplicit) (Syntax: '$""literal{c,1:format}""')
                            IFlowCaptureReferenceOperation: 2 (IsInitialization) (OperationKind.FlowCaptureReference, Type: null, IsImplicit) (Syntax: '$""literal{c,1:format}""') // This is the new node
                            InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                            OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                      Initializer:
                        null
```
## Alternative Designs

I opted for `IsInitialization` because flow captures aren't really "declared" per se. However, I could switch to `IsDeclared` in that API if we think the naming consistency is more important.

## Risks

There is the potential that some DFA analysis is expecting to see an explicit `IFlowCaptureOperation` before encountering an `IFlowCaptureReferenceOperation`. @mavasani to advise on this risk.

**Comment by 333fred** [comment]
## API Review Notes

We rename `IsInitialization` to `IsCapture` for consistency with the `IFlowCaptureOperation` name. Otherwise, the API is approved.

**Comment by 333fred** [comment]
Implemented in #57483.
